test/reverseRegistrar/TestL2ReverseRegistrar.js

```diff
@@ -0,0 +1,289 @@
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const { namehash } = require('../test-utils/ens')
+const { EMPTY_ADDRESS } = require('../test-utils/constants')
+
+describe('L2ReverseRegistrar', function () {
+  let L2ReverseRegistrar
+  let L2ReverseRegistrarWithAccount2
+  let MockSmartContractWallet
+  let MockOwnable
+  let signers
+  let account
+  let account2
+  let setNameForAddrWithSignatureFuncSig =
+    'setNameForAddrWithSignature(address,string,uint256,bytes)'
+  let setNameForAddrWithSignatureAndOwnableFuncSig =
+    'setNameForAddrWithSignatureAndOwnable(address,address,string,uint256,bytes)'
+  let setTextForAddrWithSignatureFuncSig =
+    'setTextForAddrWithSignature(address,string,string,uint256,bytes)'
+  let setTextForAddrWithSignatureAndOwnableFuncSig =
+    'setTextForAddrWithSignatureAndOwnable(address,address,string,string,uint256,bytes)'
+
+  before(async function () {
+    signers = await ethers.getSigners()
+    account = await signers[0].getAddress()
+    account2 = await signers[1].getAddress()
+
+    const L2ReverseRegistrarFactory = await ethers.getContractFactory(
+      'L2ReverseRegistrar',
+    )
+    L2ReverseRegistrar = await L2ReverseRegistrarFactory.deploy(
+      namehash('optimism.reverse'),
```

Comment:
Are we using `optimism.reverse` instead of cointype?

Reply:
No, but it was just an example since we it's a test.